docs/fsharp/tutorials/asynchronous-and-concurrent-programming/async.md

```diff
@@ -1,213 +1,349 @@
+ms.date: 12/17/2018
 ---
 # Async Programming in F# #
 
-> [!NOTE]
-> Some inaccuracies have been discovered in this article.  It is being rewritten.  See [Issue #666](https://github.com/dotnet/docs/issues/666) to learn about the changes.
+Asynchronous programming is essential to modern applications. The applications for asynchronous programming are large in number. There are two primary use cases that most developers will encounter:
 
-Async programming in F# can be accomplished through a language-level programming model designed to be easy to use and natural to the language.
+* Having a non-blocking server process that can service incoming requests while awaiting results for other requests
+* Having a responsive UI or main thread while background work is happening
 
-The core of async programming in F# is `Async<'T>`, a representation of work that can be triggered to run in the background, where `'T` is either the type returned via the special `return` keyword or `unit` if the async workflow has no result to return.
+Although background work may involve multiple threads, there is no affinity between threading and asynchrony.
 
-The key concept to understand is that an async expression’s type is `Async<'T>`, which is merely a _specification_ of work to be done in an asynchronous context. It is not executed until you explicitly start it with one of the starting functions (such as `Async.RunSynchronously`). Although this is a different way of thinking about doing work, it ends up being quite simple in practice.
+## Asynchrony defined
 
-For example, say you wanted to download the HTML from dotnetfoundation.org without blocking the main thread. You can accomplish it like this:
+If you look at the etymology of the word "asynchronous", there are two pieces:
+
+* "a", meaning "not"
+* "synchronous", meaning "at the same time"
+
+Put together, "asynchronous" simply means, "not at the same time". This holds true for asynchronous programming in F#. Code that executes asynchronously is code that does not necessarily execute at the same time.
+
+In other words, asynchronous computations execute independently of the main program flow. This does not imply any particular threading model, nor does it imply that a computation happens in the background. In fact, asynchronous computations could even execute synchronously, depending on the nature of the computation and the environment the computation is executing in.
+
+The main takeaway you should have is that because asynchronous computations are simply independent of the main program flow, there are few guarantees about when or how a computation executes. This necessitates a means to control their execution. In F#, there are multiple ways to do this.
+
+## Core concepts
+
+In F#, asynchronous programming is centered around three core components:
+
+* The `Async<'T>` type, which represents an asynchronous computation
```

Comment:
I like how this is succinct but would be very tempted to load on the fact that it's a builder affording one the ability to run a computation

   The `Async<'T>` type, which represents a Builder for a given computation from which one can compose a computation flow

i.e. allude to composition and workflows. `Task<T>` also "represents an asynchronous computation"